packages/iTwinUI-react/src/core/Table/actionHandlers/selectHandler.ts

```diff
@@ -56,6 +56,91 @@ export const onSelectHandler = <T extends Record<string, unknown>>(
   onSelect?.(selectedData, newState);
 };
 
+export const onShiftSelectHandler = <T extends Record<string, unknown>>(
+  state: TableState<T>,
+  action: ActionType,
+  instance: TableInstance<T>,
+) => {
+  const resetSelection = () => {
+    instance.rows.forEach((r) => r.toggleRowSelected(false));
+  };
+
+  const getGreaterId = (id: string) => {
+    const greaterId = id.split('.');
+    greaterId[greaterId.length - 1] = `${
+      parseInt(greaterId[greaterId.length - 1]) + 1
+    }`;
+    return greaterId.join('.');
+  };
+
+  const getParentId = (id: string) => {
+    const lastDotIndex = id.lastIndexOf('.');
+    if (lastDotIndex < 0) {
+      return null;
+    }
+    return id.substring(0, lastDotIndex);
+  };
+
+  // Returns true if endId is a parent of currentId
+  // E.g. currentId = 1.4.2.3, endId = 1.4  -> true
+  // E.g. currentId = 1.2.5, endId = 1.3    -> false
+  const isAncestorParent = (currentId: string, endId: string) => {
+    return endId.indexOf(currentId) === 0;
+  };
```

Comment:
Is `isAncestorParent()` a good name for this function? Can you suggest a better name?

Reply:
i can't think of a better name but it doesn't really matter since it's an internal helper function. All i would suggest is to convert the comment to a JSDoc comment. This way it will show up when you hover this function name anywhere in the code.

```suggestion
  /**
    * Returns true if endId is a parent of currentId
    * E.g. currentId = 1.4.2.3, endId = 1.4  -> true
    * E.g. currentId = 1.2.5, endId = 1.3    -> false
    */
  const isAncestorParent = (currentId: string, endId: string) => {
    return endId.indexOf(currentId) === 0;
  };
```

Reply:
**[DONE]** Ohh, nice to know about JSDocs, thanks